src/index.js

```diff
@@ -1,50 +1,76 @@
 import { render } from 'react-dom';
 import React, { Component } from 'react';
-import { HashRouter, Route} from 'react-router-dom';
+import { HashRouter, Route } from 'react-router-dom';
 import { Provider, connect } from 'react-redux';
-const root = document.querySelector('#root')
+const root = document.querySelector('#root');
 
-import Nav from './Nav';
-import store, {getUsersThunk, getProductsThunk} from './store';
-import Users from './Users';
-import User from './SignUp';
-import Products from './Products';
-import Home from './Home';
-import Cart from './Cart';
-import { attemptSession } from './store';
+import Nav from './Components/Nav';
+import store, {
+  getUsersThunk,
+  getProductsThunk,
+  getCartThunk,
+  getOrdersThunk
+} from './redux/store';
+import Users from './Components/Users';
+import SignUp from './Components/SignUp';
+import UserProfile from './Components/UserProfile';
+import UpdateUserForm from './Components/UpdateUserForm';
+import DeleteUser from './Components/DeleteUser';
+import Order from './Components/Order';
+import Products from './Components/Products';
+import Home from './Components/Home';
+import Cart from './Components/Cart';
+import { attemptSession } from './redux/store';
 
-
-class _App extends Component{
-  componentDidMount(){
-    this.props.getUsers();
-    this.props.getProducts();
-    this.props.attemptSession()
-      .catch(ex => console.log(ex))
+class _App extends Component {
+  componentDidMount() {
+    const { attemptSession, getUsers, getProducts, getCart, getOrders } = this.props;;
+    attemptSession().catch(() => { throw new Error('not logged in') });
+    getUsers();
```

Comment:
Essentially your guys plan for loading data is to load ALL of the data when we first load up the app, while this works momentarily, it probably isn't the best solution. We want components that **need** the data to be the ones to load the data. That way we only ask for the information that needs to be displayed, when its being displayed.